plugins/cpp/service/src/cppservice.cpp

```diff
@@ -1611,14 +1611,20 @@ CppServiceHandler::getTags(const std::vector<model::CppAstNode>& nodes_)
             tags[node.id].push_back(visibility);
         }
 
-        //--- Virtual Tag ---//
+        //--- Other Tags ---//
 
         FuncResult funcNodes = _db->query<cc::model::CppFunction>(
           FuncQuery::entityHash == defNode.entityHash);
-        const model::CppFunction& funcNode = *funcNodes.begin();
+        if (!funcNodes.empty())
+        {
+          const model::CppFunction& funcNode = *funcNodes.begin();
 
-        for (const model::Tag& tag : funcNode.tags)
-          tags[node.id].push_back(model::tagToString(tag));
+          for (const model::Tag& tag : funcNode.tags)
+            tags[node.id].push_back(model::tagToString(tag));
+        }
+        else
+          LOG(warning) << "Database query result was not expected to be empty. "
+                       << __FILE__ << ", line #" << __LINE__;
```

Comment:
I'd rely on the log message being legible to human beings and unique enough to be searched if needed (i.e. "Database query result".. which query? what database?) instead of relying on `__FILE__` and `__LINE__` being defined in the preprocessor stack at all (as opposed to `std::source_location` and its "backend", `__builtin_line()` which is also conditionally portable) and especially being defined to the value that is legible...

Seeing `/var/lib/ci/worker/job/89/workspace/foo/bar/baz/qux.cpp, line# 43892983` won't really tell the user it is an issue with what exact query in the C++ plugin...

(Moreover, all these additional strings just increase the size of the generated library, even in a release build.)

Reply:
"Unexpected empty result when querying tags of C++ function [function name]"?